From a54fdc19b548a24fa6b58a8352ea8bd1e64da53b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20Hru=C5=A1eck=C3=BD?= <michal.hrusecky@turris.com>
Date: Fri, 10 Dec 2021 15:15:44 +0100
Subject: [PATCH] rescue-image: initrd.gz is not gzip, but cpio

1. initrd.gz: ASCII cpio archive (SVR4 with no CRC)
Replace it by ugly cp instead of renaming it in two packages
---
 hardware/rescue-image/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hardware/rescue-image/Makefile b/hardware/rescue-image/Makefile
index 2254fffdf..162340e9c 100644
--- a/hardware/rescue-image/Makefile
+++ b/hardware/rescue-image/Makefile
@@ -103,7 +103,7 @@ define Build/Compile
 	cd $(LINUX_DIR); ./usr/gen_initramfs.sh \
 	-o "$(PKG_BUILD_DIR)"/kernel/linux-$(LINUX_VERSION)/arch/$(LINUX_KARCH)/boot/initrd.gz -u squash -g squash \
 	"$(PKG_BUILD_DIR)/rootfs" "$(PKG_BUILD_DIR)/rootfs.extra"
-	gzip -d "$(PKG_BUILD_DIR)"/kernel/linux-$(LINUX_VERSION)/arch/$(LINUX_KARCH)/boot/initrd.gz
+	cp "$(PKG_BUILD_DIR)"/kernel/linux-$(LINUX_VERSION)/arch/$(LINUX_KARCH)/boot/initrd.gz "$(PKG_BUILD_DIR)"/kernel/linux-$(LINUX_VERSION)/arch/$(LINUX_KARCH)/boot/initrd
 	$(CP) "$(PKG_BUILD_DIR)"/rescue.its "$(PKG_BUILD_DIR)"/kernel/linux-$(LINUX_VERSION)/arch/$(LINUX_KARCH)/boot/rescue.its
 endef
 
-- 
2.32.0

